src/r2mm/downloading/BetterThunderstoreDownloader.ts

```diff
@@ -86,35 +87,23 @@ export default class BetterThunderstoreDownloader extends ThunderstoreDownloader
                            ignoreCache: boolean,
                            callback: (progress: number, modName: string, status: number, err: R2Error | null) => void,
                            completedCallback: (modList: ThunderstoreCombo[]) => void) {
+
+        const modList = await ProfileModList.getModList(profile);
+        if (modList instanceof R2Error) {
+            return callback(0, mod.getName(), StatusEnum.FAILURE, modList);
+        }
+
         let dependencies: ThunderstoreCombo[] = [];
         await this.buildDependencySet(modVersion, dependencies, DependencySetBuilderMode.USE_EXACT_VERSION);
         this.sortDependencyOrder(dependencies);
         const combo = new ThunderstoreCombo();
```

Comment:
Possible idea for future PR or another commit: this method could take a combo object as the argument instead of `mod` & `modVersion`. All the other methods in the provider already deal with combos, so that would be consistent.

Reply:
I did this change now, seemed like a simple change that removes extra lines.